Length (Spectral) nodes and QueueStore

mhusinsky opened the pull request:
Here are some new nodes I'd like to contribute. I implemented them generically and followed the instructions in the wiki, yet I'm not sure if I met all criteria. Please double check.

I'm thankful for any comments and suggestions.

mhusinsky committed:
Added Length (2d Vector Spectral), Length (3d Vector Spectral) and QueueStore (Spreads)

azeno commented:
Well looks good except for the naming of one input on the QueueStore node -> always use upper case for first character.

mhusinsky committed:
* changed behaviour of QueueStore so it does not automatically create a new queue for each newly inserted data after an "insert pause"

* renamed pins to more senseful names
** Split -> New
** Duplicated on split -> Duplicate Last Input
** Frames Recorded -> Output Queue Size
* added QueueStore to generics template file

mhusinsky committed:
* renamed class names so they fit in the generics template naming scheme

* created SpectralNodesTemplates

mhusinsky commented:
fixed the pin naming and did some minor internal changes which make sense and should have been there in the first place.

mhusinsky committed:
* Added "Append" pin that inserts the queue at the last position of the group

* cleanup

azeno commented:
Continued by https://github.com/vvvv/vvvv-sdk/pull/252
